script/bump-version.py

```diff
@@ -11,28 +12,76 @@
 
 
 def main():
-  if len(sys.argv) != 2 or sys.argv[1] == '-h':
-    print 'Usage: bump-version.py [<version> | major | minor | patch]'
+
+  parser = argparse.ArgumentParser(
+    description='Bump version numbers. Must specify at least one option.'
+  )
+  parser.add_argument(
+    '--version', 
+    default=None, 
+    dest='new_version', 
+    help='new version number'
+  )
+  parser.add_argument(
+    '--bump', 
+    action='store', 
+    default=None, 
+    dest='bump', 
+    help='increment [major | minor | patch | beta]'
+  )
+  parser.add_argument(
+    '--stable', 
+    action='store_true', 
+    default= False, 
+    dest='stable', 
+    help='promote to stable (i.e. remove `-beta.x` suffix)'
+  )
+  parser.add_argument(
+    '--dry-run', 
+    action='store_true', 
+    default= False, 
+    dest='dry_run', 
+    help='just to check that version number is correct'
+  )
+
+  args = parser.parse_args()
+
+  if args.new_version == None and args.bump == None and args.stable == False:
+    parser.print_help()
     return 1
 
-  option = sys.argv[1]
-  increments = ['major', 'minor', 'patch', 'build']
-  if option in increments:
-    version = get_electron_version()
-    versions = parse_version(version.split('-')[0])
-    versions = increase_version(versions, increments.index(option))
-  else:
-    versions = parse_version(option)
+  increments = ['major', 'minor', 'patch', 'beta']
+  
+  curr_version = get_electron_version()
+  versions = parse_version(re.sub('-beta', '', curr_version))
+  
+  if args.bump in increments:
+    versions = increase_version(versions, increments.index(args.bump))
+    if versions[3] == '0':
+      # beta starts at 1
```

Comment:
> beta starts at 1

So the first 1.8.5 beta would be `1.8.5-beta1`? If so, we should update #10336 to reflect this.

Reply:
Yup. That's to adhere to the format restriction of the `FILEVERSION` number for windows .exe, which requires 4 numbers at all times.